pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -0,0 +1,350 @@
+/*
+Copyright 2020 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grafeas
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/tektoncd/chains/pkg/chains/formats"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	attestationpb "github.com/grafeas/grafeas/proto/v1beta1/attestation_go_proto"
+	commonpb "github.com/grafeas/grafeas/proto/v1beta1/common_go_proto"
+	pb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	"github.com/tektoncd/chains/pkg/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	gstatus "google.golang.org/grpc/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logtesting "knative.dev/pkg/logging/testing"
+)
+
+type args struct {
+	tr           *v1beta1.TaskRun
+	trPayload    []byte
+	trSignature  string
+	trOpts       config.StorageOpts
+	ociPayload   []byte
+	ociSignature string
+	ociOpts      config.StorageOpts
+}
+
+type testConfig struct {
+	name    string
+	args    args
+	wantErr bool
```

Comment:
I don't think this is used right now, but it would be good to add test cases for common error cases.

Reply:
add test for tekton format (which should return error because we don't want grafeas backend to support tekton format payload.) 820eac28cf421f647b4d9b6b702409c928735bff